python/lambdas/copy-files-from-tdr/lambda_function.py

```diff
@@ -1,85 +1,97 @@
 
-    try:
-        s3_client.head_object(source_bucket, f"{file_id}.metadata")
-    except botocore.exceptions.ClientError as ex:
-        raise Exception(f"Object {file_id}.metadata does not exist, underlying error is: {ex}")
+def assert_objects_exist_in_bucket(source_bucket, files):
+    for file in files:
+        try:
+            s3_client.head_object(source_bucket, file)
+        except botocore.exceptions.ClientError as ex:
+            raise Exception(f"Object '{file}' does not exist in '{source_bucket}', underlying error is: '{ex}'")
 
 
-def validate_formats(bucket, s3_key):
+def validate_metadata(bucket, s3_key):
     response = s3_client.get_object(Bucket=bucket, Key=s3_key)
-    json_content = json.loads(response['Body'])
-    uuid_content = json_content['UUID']
-    try:
-        uuid_object = uuid.UUID(uuid_content)
-    except ValueError:
-        raise Exception(f"Unable to parse UUID, '{uuid_content}'. Invalid format")
+    validate_mandatory_fields_exist(response)
+    validate_formats(response, bucket, s3_key)
 
-    transfer_initiated_date_content = json_content['TransferInitiatedDatetime']
-    try:
-        transfer_initiated_date =  datetime.strptime(transfer_initiated_date_content, "%Y-%m-%d %H:%M:%S")
-    except ValueError:
-        raise Exception(f"Unable to parse date, '{transfer_initiated_date_content}'. Invalid format")
 
-    series = json_content['Series']
-    if not series.strip():
-        raise Exception(f"Empty series value, unable to proceed")
+def validate_mandatory_fields_exist(get_object_response):
+    json_metadata = json.loads(get_object_response['Body'])
+    try:
+        validate(json_metadata, metadata_schema)
+    except jsonschema.exceptions.ValidationError as err:
+        raise Exception(err.message)
 
     return True
 
-def validate_metadata(bucket, s3_key):
-    validate_mandatory_fields_exist(bucket, s3_key)
-    validate_formats(bucket, s3_key)
 
-def validate_mandatory_fields_exist(bucket, s3_key):
-    response = s3_client.get_object(Bucket=bucket, Key=s3_key)
-    json_content = response['Body']
-    data = json.loads(json_content)
+def validate_formats(get_object_response, bucket, s3_key):
+    json_metadata = json.loads(get_object_response['Body'])
+    uuid_content = json_metadata['UUID']
     try:
-        validate(data, incoming_schema)
-    except jsonschema.exceptions.ValidationError as err:
-        raise Exception(err.message)
+        uuid.UUID(uuid_content)
+    except ValueError:
+        raise Exception(
+            f"Unable to parse UUID, '{uuid_content}' from file '{s3_key}' in bucket '{bucket}'. Invalid format")
+
+    series = json_metadata['Series']
+    if not series.strip():
+        raise Exception(f"Empty Series value in file '{s3_key}' in bucket '{bucket}'. Unable to proceed")
+
+    #FIXME, Can we use this regex pattern? The test series MOCK1 123 fails this pattern
+    # pattern = "^([A-Z]{1,4} [1-9][0-9]{0,3})"
```

Comment:
1. I know this is a work in progress but it needs a `$` at the end. 
2. also, I [upgraded the regex](https://github.com/nationalarchives/dr2-ingest/blob/main/scala/lambdas/ingest-validate-generic-ingest-inputs/src/main/resources/metadata-validation-schemas/archive-and-content-folder-validation-schema.json) in the validator that I did to include `MOCK1 123`

Reply:
Done